# Situacion Problema

* Lo primero que haremos es importar e instalar las bibliotecas que haran falta para poder
  responder las preguntas que se haran posteriormente

In [2]:
pip install statsmodels

     |████████████████████████████████| 9.8 MB 4.9 MB/s            
     |████████████████████████████████| 233 kB 60.3 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [3]:
pip install seaborn

     |████████████████████████████████| 292 kB 4.7 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [4]:
pip install sklearn 

  Preparing metadata (setup.py) ... done
     |████████████████████████████████| 23.2 MB 4.7 MB/s            ██████████████▏                | 11.0 MB 4.7 MB/s eta 0:00:03
     |████████████████████████████████| 306 kB 57.9 MB/s            
  Created wheel for sklearn: filename=sklearn-0.0-py2.py3-none-any.whl size=1309 sha256=942f3aa5589515f5c61185524e060659562a59d1edc54467b202b36f4bd886da
  Stored in directory: /home/jovyan/.cache/pip/wheels/46/ef/c3/157e41f5ee1372d1be90b09f74f82b10e391eaacca8f22d33e
Successfully built sklearn
Note: you may need to restart the kernel to use updated packages.


In [5]:
import numpy as np
import pandas as pd
import seaborn as sns; sns.set()
import matplotlib.pyplot as plt
import matplotlib as mpl
import statsmodels.api as sm
from sklearn import linear_model
from sklearn.linear_model import LinearRegression 

* Ahora se procedera a bucar la base de datos dento de los archivos de Github

In [6]:
%ls SP/

COVID-19_Vaccinations_in_the_United_States_County.csv


* Se leera la base de datos

In [7]:
df = pd.read_csv('SP/COVID-19_Vaccinations_in_the_United_States_County.csv') 
df.head()

Date     FIPS  MMWR_week        Recip_County Recip_State  \
0  11/30/21   6077.0       48.0  San Joaquin County          CA   
1  11/30/21  26013.0       48.0       Baraga County          MI   
2  11/30/21  16071.0       48.0       Oneida County          ID   
3  11/30/21  38001.0       48.0        Adams County          ND   
4  11/30/21  20071.0       48.0      Greeley County          KS   

   Series_Complete_Pop_Pct  Series_Complete_Yes  Series_Complete_12Plus  \
0                     55.7             424244.0                423613.0   
1                     56.6               4650.0                  4649.0   
2                     38.0               1722.0                     NaN   
3                     45.8               1014.0                  1014.0   
4                     48.5                598.0                   598.0   

   Series_Complete_12PlusPop_Pct  Series_Complete_18Plus  ...  SVI_CTGY  \
0                           67.4                390835.0  ...         D   
1                           63.3                  4533.0  ...         C   
2                            NaN                  1716.0  ...         B   
3                           52.4                   985.0  ...         A   
4                           59.5                   571.0  ...         C   

   Series_Complete_Pop_Pct_SVI  Series_Complete_12PlusPop_Pct_SVI  \
0                         16.0                               16.0   
1                         12.0                               12.0   
2                          6.0                                7.0   
3                          3.0                                4.0   
4                         11.0                               12.0   

   Series_Complete_18PlusPop_Pct_SVI  Series_Complete_65PlusPop_Pct_SVI  \
0                               16.0                               16.0   
1                               12.0                               12.0   
2                                8.0                                8.0   
3                                4.0                                4.0   
4                               12.0                               12.0   

   Metro_status  Series_Complete_Pop_Pct_UR_Equity  \
0         Metro                                4.0   
1     Non-metro                                8.0   
2     Non-metro                                6.0   
3     Non-metro                                7.0   
4     Non-metro                                7.0   

   Series_Complete_12PlusPop_Pct_UR_Equity  \
0                                      4.0   
1                                      8.0   
2                                      7.0   
3                                      8.0   
4                                      8.0   

   Series_Complete_18PlusPop_Pct_UR_Equity  \
0                                      4.0   
1                                      8.0   
2                                      8.0   
3                                      8.0   
4                                      8.0   

   Series_Complete_65PlusPop_Pct_UR_Equity  
0                                      4.0  
1                                      8.0  
2                                      8.0  
3                                      8.0  
4                                      8.0  

[5 rows x 32 columns]

* Para poder crear las preguntas se hara un listado de los datos que tiene esta base de datos

In [8]:
df.columns

Index(['Date', 'FIPS', 'MMWR_week', 'Recip_County', 'Recip_State',
       'Series_Complete_Pop_Pct', 'Series_Complete_Yes',
       'Series_Complete_12Plus', 'Series_Complete_12PlusPop_Pct',
       'Series_Complete_18Plus', 'Series_Complete_18PlusPop_Pct',
       'Series_Complete_65Plus', 'Series_Complete_65PlusPop_Pct',
       'Completeness_pct', 'Administered_Dose1_Recip',
       'Administered_Dose1_Pop_Pct', 'Administered_Dose1_Recip_12Plus',
       'Administered_Dose1_Recip_12PlusPop_Pct',
       'Administered_Dose1_Recip_18Plus',
       'Administered_Dose1_Recip_18PlusPop_Pct',
       'Administered_Dose1_Recip_65Plus',
       'Administered_Dose1_Recip_65PlusPop_Pct', 'SVI_CTGY',
       'Series_Complete_Pop_Pct_SVI', 'Series_Complete_12PlusPop_Pct_SVI',
       'Series_Complete_18PlusPop_Pct_SVI',
       'Series_Complete_65PlusPop_Pct_SVI', 'Metro_status',
       'Series_Complete_Pop_Pct_UR_Equity',
       'Series_Complete_12PlusPop_Pct_UR_Equity',
       'Series_Complete_18PlusPop_P

# Se utilizaran los datos:
* Recip_State = "Recipient State", 
* Series_Complete_Pop_Pct = "Percent of people who are fully vaccinated (have second dose of a two-dose vaccine or one dose of a single-dose vaccine) based on the jurisdiction and county where recipient lives", 
* Series_Complete_12PlusPop_Pct = "Percent of people 12+ who are fully vaccinated (have second dose of a two-dose vaccine or one dose of a single-dose vaccine) based on the jurisdiction where recipient lives"
* Series_Complete_18PlusPop_Pct = "Percent of people 18+ who are fully vaccinated (have second dose of a two-dose vaccine or one dose of a single-dose vaccine) based on the jurisdiction and county where recipient lives"
* Series_Complete_65PlusPop_Pct = "Percent of people 65+ who are fully vaccinated (have second dose of a two-dose vaccine or one dose of a single-dose vaccine) based on the jurisdiction where recipient lives"
* Series_Complete_Yes = "Total number of people who are fully vaccinated (have second dose of a two-dose vaccine or one dose of a single-dose vaccine) based on the jurisdiction and county where recipient lives"


# Preguntas que se contestaran en esta situacion problema
1- `Cuales son los condados con mayor porcentaje de personas vacunadas completamente?`



In [9]:
V = ['Recip_County','Series_Complete_Pop_Pct']
dfv = df[V]
df_p = dfv.sort_values(by='Series_Complete_Pop_Pct', ascending=False,axis=0, inplace=False)
df_p.head()

Recip_County  Series_Complete_Pop_Pct
84    Lares Municipio                     83.9
97      Blaine County                     75.1
57  Maunabo Municipio                     67.8
64  Burlington County                     67.7
20      Nelson County                     65.1

Podemos observa que los 5 condados con mayor porcentaje de vacunados por completo son  con 83.9%, Blaine County con 75.1%, Maunabo Municipio con 67.8%, Burlington County con 67.7%, Nelson County con 65.1%

2- `Cuales estados con mayor porcentaje de poblacion completamente vacunada?`

In [43]:
dp = pd.pivot_table(df, index = 'Recip_State', values = 'Series_Complete_Pop_Pct')
dp

Series_Complete_Pop_Pct
Recip_State                         
AL                         41.050000
AR                         41.950000
AZ                         51.500000
CA                         27.850000
FL                         57.600000
GA                         21.900000
IA                         50.200000
ID                         38.000000
IL                         49.625000
IN                         41.800000
KS                         48.500000
KY                         43.600000
MD                         62.700000
MI                         60.050000
MN                         63.350000
MO                         37.020000
MS                         40.900000
MT                         52.366667
NC                         44.700000
ND                         44.725000
NE                         34.780000
NJ                         67.700000
NM                         54.700000
NY                         54.600000
OH                         47.400000
OK                         46.975000
OR                         55.900000
PA                         41.550000
PR                         70.666667
SC                         28.400000
SD                         49.800000
TN                         39.800000
TX                         45.728571
UT                         44.000000
VA                         48.425000
VT                         50.150000
WA                         55.800000
WI                         50.150000
WV                         24.900000
WY                         41.900000

In [44]:
dp.max()

Series_Complete_Pop_Pct    70.666667
dtype: float64

Descubrimos que el mayor pocentaje dentro de los estados es de 70.66% y es del estado de Puerto Rico


3- `Cual es la parte de la poblicion dentro de este estado que tiene mayor porcentaje de vacunacion por completo?`

In [49]:
Estado = df['Recip_State'] == 'PR'
dE = df[Estado]
dE

Date     FIPS  MMWR_week       Recip_County Recip_State  \
57  11/30/21  72095.0       48.0  Maunabo Municipio          PR   
84  11/30/21  72081.0       48.0    Lares Municipio          PR   
90  11/30/21  72063.0       48.0   Gurabo Municipio          PR   

    Series_Complete_Pop_Pct  Series_Complete_Yes  Series_Complete_12Plus  \
57                     67.8               7001.0                  7001.0   
84                     83.9              20356.0                 20350.0   
90                     60.3              28388.0                 28376.0   

    Series_Complete_12PlusPop_Pct  Series_Complete_18Plus  ...  SVI_CTGY  \
57                           76.1                  6500.0  ...         C   
84                           94.0                 18744.0  ...         D   
90                           67.6                 26234.0  ...         A   

    Series_Complete_Pop_Pct_SVI  Series_Complete_12PlusPop_Pct_SVI  \
57                         12.0                               12.0   
84                         16.0                               16.0   
90                          4.0                                4.0   

    Series_Complete_18PlusPop_Pct_SVI  Series_Complete_65PlusPop_Pct_SVI  \
57                               12.0                               12.0   
84                               16.0                               16.0   
90                                4.0                                3.0   

    Metro_status  Series_Complete_Pop_Pct_UR_Equity  \
57         Metro                                4.0   
84         Metro                                4.0   
90         Metro                                4.0   

    Series_Complete_12PlusPop_Pct_UR_Equity  \
57                                      4.0   
84                                      4.0   
90                                      4.0   

    Series_Complete_18PlusPop_Pct_UR_Equity  \
57                                      4.0   
84                                      4.0   
90                                      4.0   

    Series_Complete_65PlusPop_Pct_UR_Equity  
57                                      4.0  
84                                      4.0  
90                                      3.0  

[3 rows x 32 columns]

In [79]:
dja = pd.pivot_table(df, index = Estado, values = 'Series_Complete_18PlusPop_Pct')
dj[1:]

Series_Complete_18PlusPop_Pct
Recip_State                               
True                             79.566667

In [77]:
da = pd.pivot_table(df, index = Estado, values = 'Series_Complete_12PlusPop_Pct')
da [1:]

Series_Complete_12PlusPop_Pct
Recip_State                               
True                             79.233333

In [74]:
dma = pd.pivot_table(df, index = Estado, values = 'Series_Complete_65PlusPop_Pct')
dma[1:]

Series_Complete_65PlusPop_Pct
Recip_State                               
True                                  83.1